src/main/scala/ai/privado/dataflow/DuplicateFlowProcessor.scala

```diff
@@ -167,46 +167,89 @@ object DuplicateFlowProcessor {
     ruleCache: RuleCache
   ): Unit = {
 
+    val expendedSourceSinkInfo = processExpendedSourceSinkData(dataflowMapByPathId, privadoScanConfig, ruleCache)
+
+    expendedSourceSinkInfo.foreach(flow => {
+      AuditCache.addIntoBeforeSecondFiltering(SourcePathInfo(flow.pathSourceId, flow.sinkId, flow.sinkPathId))
+      filterFlowsOverlappingWithOtherDataElement(
+        flow.pathSourceId,
+        flow.dataflowsMapByType,
+        flow.dataflowSinkType,
+        flow.dataflowNodeType,
+        ruleCache,
+        flow.sinkId,
+        flow.sinkPathId
+      )
+    })
+  }
+
+  case class ExpendedSourceSinkInfo(
+    pathSourceId: String,
+    sinkId: String,
+    dataflowSinkType: String,
+    dataflowNodeType: String,
+    dataflowsMapByType: Map[String, Path],
```

Comment:
@ankit-privado Storing the complete map is not a good idea, as for each item you will be storing this map, which will be memory intensive